### DISCUSSION TITLE
Is there any mongoose library that can introspect collection and turn into mongoose Schema

### DISCUSSION THREAD

**Discussion by masterbater** [Automattic/mongoose#14044] [Q&A]
Like prisma db pull it reads your collection and turn into schema, where it helps save time then modify if there are some inconsistencies

**Comment by vkarpov15** [comment]
Not currently. It is tricky to do this with MongoDB in general because database doesn't have schema by default.

**Comment by Mihailoff** [comment]
MongoDB compass can analyze schema https://www.mongodb.com/docs/compass/current/schema/#analyze-your-data-schema

It is open source https://github.com/mongodb-js/compass/blob/main/packages/compass-schema/src/modules/schema-analysis.ts

And it uses this library https://github.com/mongodb-js/mongodb-schema

Based on what `mongodb-schema` can do, it seems possible to convert results into some basic Mongoose schema.

**Comment by mmadson** [comment]
Prisma did exactly what @Mihailoff suggested for their mongo support. See article here:

https://www.mongodb.com/developer/products/mongodb/how-prisma-introspects-a-schema-from-a-mongodb-database/
